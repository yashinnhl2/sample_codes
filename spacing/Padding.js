import React from 'react';
import styled from 'styled-components';
import { p, pb, pl, pr, pt, px, py } from './core';

const StyledPadding = styled.div`
  ${({ inline }) => inline && 'display: inline-block !important;'}
  ${({ all }) => p(all)}
  ${({ horizontal }) => px(horizontal)}
  ${({ vertical }) => py(vertical)}
  ${({ top }) => pt(top)}
  ${({ right }) => pr(right)}
  ${({ bottom }) => pb(bottom)}
  ${({ left }) => pl(left)}
`;

/**
 * @typedef {import('./core').ViewportProps} ViewportProps
 */

/**
 * @typedef {import('./core').CoreSizes} PaddingSize
 */

/**
 * @typedef {object} PaddingProps Padding Component Properties
 * @prop {boolean} [inline] Display the container with inline-block
 * @prop {ViewportProps|PaddingSize} [all] Padding on all sides
 * @prop {ViewportProps|PaddingSize} [vertical] Top and Bottom padding
 * @prop {ViewportProps|PaddingSize} [horizontal] Left and Right padding
 * @prop {ViewportProps|PaddingSize} [top] Top padding
 * @prop {ViewportProps|PaddingSize} [bottom] Bottom padding
 * @prop {ViewportProps|PaddingSize} [left] Left padding
 * @prop {ViewportProps|PaddingSize} [right] Right padding
 */

/**
 * A container with configurable responsive padding
 * @type {React.FC<PaddingProps>}
 * @param {PaddingProps} props
 * @returns
 */
const Padding = props => <StyledPadding {...props} />;

export { Padding };
