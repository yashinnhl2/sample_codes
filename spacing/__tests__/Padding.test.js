import React from 'react';
import matchesSnapshot from '@upgrade/ui-utils/utils/assertions/matches-snapshot';
import { Padding } from '../Padding';

describe('Padding', () => {
  it('Should have no style when no props are passed', () => {
    matchesSnapshot(<Padding />);
  });

  it('Should have inline props if inline prop is passed', () => {
    matchesSnapshot(<Padding inline />);
  });

  it('Should have xs to rem Padding from all side', () => {
    matchesSnapshot(<Padding all={{ xs: 'xs' }} />);
  });

  it('Should have xs, sm to rem Padding from all side', () => {
    matchesSnapshot(<Padding all={{ xs: 'xs', sm: 'md' }} />);
  });

  it('Should have xs, sm to rem Padding from right side only', () => {
    matchesSnapshot(<Padding right={{ xs: 'md', sm: 'lg' }} />);
  });

  it('Should have Padding top only without any media queries', () => {
    matchesSnapshot(<Padding top="2xl" />);
  });

  it('Should have Padding left only without any media queries', () => {
    matchesSnapshot(<Padding left="2xl" />);
  });

  it('Should have Padding left only without any media queries, if user pass string', () => {
    matchesSnapshot(<Padding right="xl" />);
  });

  it('Should have no style, if user pass empty string', () => {
    matchesSnapshot(<Padding left="" />);
  });

  it('Should have empty value, if user pass empty object', () => {
    matchesSnapshot(<Padding left={{}} />);
  });
});
