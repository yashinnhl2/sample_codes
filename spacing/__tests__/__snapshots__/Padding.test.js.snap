// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Padding Should have Padding left only without any media queries 1`] = `
.c0 {
  padding-left: 3rem;
}

<div
  className="c0"
/>
`;

exports[`Padding Should have Padding left only without any media queries, if user pass string 1`] = `
.c0 {
  padding-right: 2rem;
}

<div
  className="c0"
/>
`;

exports[`Padding Should have Padding top only without any media queries 1`] = `
.c0 {
  padding-top: 3rem;
}

<div
  className="c0"
/>
`;

exports[`Padding Should have empty value, if user pass empty object 1`] = `
<div
  className=""
/>
`;

exports[`Padding Should have inline props if inline prop is passed 1`] = `
.c0 {
  display: inline-block !important;
}

<div
  className="c0"
/>
`;

exports[`Padding Should have no style when no props are passed 1`] = `
<div
  className=""
/>
`;

exports[`Padding Should have no style, if user pass empty string 1`] = `
<div
  className=""
/>
`;

exports[`Padding Should have xs to rem Padding from all side 1`] = `
.c0 {
  padding: 0.25rem;
}

<div
  className="c0"
/>
`;

exports[`Padding Should have xs, sm to rem Padding from all side 1`] = `
.c0 {
  padding: 0.25rem;
}

@media (min-width:48em) {
  .c0 {
    padding: 1rem;
  }
}

<div
  className="c0"
/>
`;

exports[`Padding Should have xs, sm to rem Padding from right side only 1`] = `
.c0 {
  padding-right: 1rem;
}

@media (min-width:48em) {
  .c0 {
    padding-right: 1.5rem;
  }
}

<div
  className="c0"
/>
`;
